src/plugins/vis_default_editor/public/components/controls/include_exclude.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect } from 'react';
+import { isArray } from 'lodash';
+import { AggParamEditorProps } from '../agg_param_props';
+import { StringParamEditor } from './string';
+import { search } from '../../../../data/public';
+import { SimpleNumberList } from './components/simple_number_list';
+const { isNumberType } = search.aggs;
+
+export function IncludeExcludeParamEditor({
+  agg,
+  aggParam,
+  value,
+  setValue,
+  setValidity,
+  showValidation,
+  setTouched,
+  editorConfig,
+  formIsTouched,
+  metricAggs,
+  schemas,
+  state,
+}: AggParamEditorProps<string | Array<number | undefined>>) {
+  const isAggOfNumberType = isNumberType(agg);
+
+  // This useEffect converts value from string type to number and back when the field type is changed
+  useEffect(() => {
+    if (isAggOfNumberType && !isArray(value) && value !== undefined) {
+      setValue(
+        value
+          .split(',')
+          .map(item => parseFloat(item))
+          .filter(number => !isNaN(number)) as number[]
+      );
```

Comment:
There is the case when a user types some string to the input, then they switch to a number type field. And this code will set `value` as an empty array. But `SimpleNumberList` will compare the value and model list (for case of discarding the changes) and it occurs that there is no the first default empty number input (which is set during initialization of models).
I suggest to set an empty string by default for such case:
```suggestion
const numberArray = value
        .split(',')
        .map(item => parseFloat(item))
        .filter(number => !isNaN(number));
 setValue(numberArray.length ? numberArray : ['']);
```